### PR TITLE
update PSP instructions for new 1.3.1 release, and separate instructions

### PR DESCRIPTION
**Pull request by cjschroder**
for older releases.

re https://github.com/SUSE/doc-cap/issues/317

### CODE REVIEW

**Comment by cjschroder** [comment]
The appendix "A.1 Pod Security Policies for SUSE Cloud Application Platform 1.3 and Older" still needs improvement re custom PSPs.

File: xml/sec_psps.xml [hunk 1]
```diff
@@ -1,33 +1,43 @@
...
+   <title>Pod Security Policies for &productname; 1.3 and Older</title>
 
    <para>
-    Currently, all the pods are created using the default
-    <literal>serviceAccount</literal> in their namespaces in order to apply the
-    <literal>unprivileged</literal> PSP. Consequently, some pods cannot be
-    created because privileged mode and privilege escalation are disabled by
-    default. (error: cannot set allowPrivilegeEscalation to
-    <literal>false</literal> and <literal>privileged</literal> to true).
+    &productname; 1.3.1 introduces inbuilt support for Pod Security
```
Comment (btat) [inline xml/sec_psps.xml]: Could we use built-in instead, thanks!

File: xml/sec_psps.xml [hunk 2]
```diff
@@ -1,33 +1,43 @@
...
+    Policies (PSPs), which are provided via &helm; charts and are set up automatically
+    without requiring manual configuration. &suse; &caasp;, Microsoft Azure 
+    Kubernetes Service (AKS), and Amazon Elastic Container Service for Kubernetes 
+    (Amazon EKS) all enable role-based access controls (RBAC) by default, and the
+    inbuilt PSPs are required for &cap; to operate.
```
Comment (btat) [inline xml/sec_psps.xml]: Could we use built-in instead, thanks!

Comment (cjschroder) [reply xml/sec_psps.xml]: yes :laughing:

File: xml/sec_psps.xml [hunk 3]
```diff
@@ -1,33 +1,43 @@
...
-<sect2
+<sect1
+ xml:id="app.psps" 
  xmlns="http://docbook.org/ns/docbook"
  xmlns:xi="http://www.w3.org/2001/XInclude"
  xmlns:xlink="http://www.w3.org/1999/xlink"
  version="5.0">
     
-   <title>&suse; &caasp; 3 PodSecurityPolicy</title>
+   <title>Pod Security Policies for &productname; 1.3 and Older</title>
```
Comment (mook-as) [inline xml/sec_psps.xml]: Possible replacement: `Manual configuration of Pod Security Policies`

File: xml/sec_psps.xml [hunk 4]
```diff
@@ -1,33 +1,43 @@
...
+    Policies (PSPs), which are provided via &helm; charts and are set up automatically
+    without requiring manual configuration. &suse; &caasp;, Microsoft Azure 
+    Kubernetes Service (AKS), and Amazon Elastic Container Service for Kubernetes 
+    (Amazon EKS) all enable role-based access controls (RBAC) by default, and the
+    built-in PSPs are required for &cap; to operate.
+</para>
+<para>
+    Older &cap; releases require manual configuration of PSPs for &cap; to operate
+    on platforms with RBACs. This section provides instructions for configuring
+    and applying the appropriate PSPs to older &cap; releases.
```
Comment (mook-as) [inline xml/sec_psps.xml]: (See above; administrators may wish to override our PSPs for other reasons too)

File: xml/sec_psps.xml [hunk 5]
```diff
@@ -1,33 +1,43 @@
...
-    To get around this, create a configuration file called
-    <filename>cap-psp-rbac.yaml</filename>. This enables both privileged mode
-    and privilege escalation. You need a running &caasp; cluster, and
-    <command>kubectl</command> configured and working. You will apply this file
-    before you deploy UAA and SCF.
+    See the upstream documentation at
+    <link xlink:href="https://kubernetes.io/docs/concepts/policy/pod-security-policy/">
+    Pod Security Policies</link>,
+    <link xlink:href="https://docs.cloudfoundry.org/concepts/roles.html"> Orgs,
+    Spaces, Roles, and Permissions</link>, and
```
Comment (mook-as) [inline xml/sec_psps.xml]: Huh, does the roles page actually say anything about PSPs? I thought that was about CF-level permissions, not Kube-level permissions (after all, CF by default doesn't have anything to do with Kube)

File: xml/sec_psps.xml [hunk 6]
```diff
@@ -1,33 +1,43 @@
...
+    <link xlink:href="https://docs.cloudfoundry.org/uaa/identity-providers.html#id-flow">
+    Identity Provider Workflow</link> for more information on understanding and 
+    using PSPs.
    </para>
-   <para>
+
+    <para>
```
Comment (mook-as) [inline xml/sec_psps.xml]: If this section is for the wholesale replacement of default PSPs, we might want to specify the ~un~nonprivileged version too.

Note that for 1.3.1+ `node-reader` cannot be overridden (we always provide that), but the PSPs are optional.

File: xml/sec_scf_values_yaml.xml [hunk 7]
```diff
@@ -0,0 +1,1894 @@
...
+ xml:id="app.scf.values.yaml" 
+ xmlns="http://docbook.org/ns/docbook"
+ xmlns:xi="http://www.w3.org/2001/XInclude"
+ xmlns:xlink="http://www.w3.org/1999/xlink"
+ version="5.0">
+
+    <title>Complete suse/scf values.yaml file</title>
+  <para>
+   This is the complete output of <command>helm inspect suse/cf</command> for
+   the current &productname; release.
```
Comment (mook-as) [inline xml/sec_scf_values_yaml.xml]: Would it make sense to specify _which_ release was current as of time of writing? (Even if this is auto-generated via a script at publishing time)

**Comment by btat** [comment]
> Process question: is there a way to view the rendered docs for PRs? (I prefer the multi-page HTML view, but any one of them is better than trying to do XML references in my head)

I could be wrong, but I don't think there is. We could/should start attaching the PDF and HTML files we build locally to the PRs though.

File: xml/cap_depl_install_caasp.xml [hunk 8]
```diff
@@ -53,7 +53,7 @@
        <listitem>
         <para>
          Release Notes:
-         <link xlink:href="https://www.suse.com/releasenotes/SUSE-CAP/1/"/>
+         <link xlink:href="https://www.suse.com/documentation/cloud-application-platform-1/index.html"/>
```
[the remark this answers is not in the document]
Comment (btat) [reply xml/cap_depl_install_caasp.xml]: Nope, but looks like https://github.com/SUSE/doc-cap/pull/335 updates the Release Notes links anyways :D

File: xml/sec_psps.xml (hunk 5 above)
Comment (cjschroder) [reply xml/sec_psps.xml]: Just passing on what I get from engineering.

File: xml/appendix_values_yaml.xml [hunk 9]
```diff
@@ -6,2171 +6,16 @@
...
-
-  # PEM-encoded certificate
-  # This value uses a generated default.
-  SAML_SERVICEPROVIDER_CERT: ~
-
-  # PEM-encoded key.
-  SAML_SERVICEPROVIDER_CERT_KEY: ~
-
-  # The password for access to the UAA database.
-  # This val
```
[the remark this answers is not in the document]
Comment (cjschroder) [reply xml/appendix_values_yaml.xml]: :) We run a daps command periodically to clean up spacing.

**Comment by cjschroder** [comment]
@btat @mook-as Please go ahead and merge this, as it's already large and messy. I will apply your feedback to a new commit.

**Comment by cjschroder** [comment]
> @btat @mook-as Please go ahead and merge this, as it's already large and messy. I will apply your feedback to a new commit.

LOL I'll just merge it, not sure what I was thinking.

File: xml/sec_scf_values_yaml.xml (hunk 7 above)
Comment (cjschroder) [reply xml/sec_scf_values_yaml.xml]: yes good idea

**Pull request merged**